hibernate-envers/src/main/java/org/hibernate/envers/configuration/internal/metadata/AuditMetadataGenerator.java

```diff
@@ -173,43 +172,12 @@ void addRevisionType(Element anyMapping, Element anyMappingEnd, boolean isKey) {
 		);
 		revTypeProperty.addAttribute( "type", "org.hibernate.envers.internal.entities.RevisionTypeType" );
 
-		// Adding the end revision, if appropriate
-		addEndRevision( anyMappingEnd );
-	}
-
-	private void addEndRevision(Element anyMapping) {
-		// Add the end-revision field, if the appropriate strategy is used.
-		if ( auditStrategy instanceof ValidityAuditStrategy ) {
-			final Element endRevMapping = (Element) revisionInfoRelationMapping.clone();
-			endRevMapping.setName( "many-to-one" );
-			endRevMapping.addAttribute( "name", verEntCfg.getRevisionEndFieldName() );
-			MetadataTools.addOrModifyColumn( endRevMapping, verEntCfg.getRevisionEndFieldName() );
-
-			anyMapping.add( endRevMapping );
-
-			if ( verEntCfg.isRevisionEndTimestampEnabled() ) {
-				// add a column for the timestamp of the end revision
-				final String revisionInfoTimestampSqlType = TimestampType.INSTANCE.getName();
-				final Element timestampProperty = MetadataTools.addProperty(
-						anyMapping,
-						verEntCfg.getRevisionEndTimestampFieldName(),
-						revisionInfoTimestampSqlType,
-						true,
-						true,
-						false
-				);
-				MetadataTools.addColumn(
-						timestampProperty,
-						verEntCfg.getRevisionEndTimestampFieldName(),
-						null,
-						null,
-						null,
-						null,
-						null,
-						null
-				);
-			}
-		}
+        // Add further columns to the aud table
+        addAdditionalColumns( anyMappingEnd );
```

Comment:
I don't think I would call this here. You end up adding a very generic method in a very specific one.

I think I would add another call in `generateMappingData()` and `CollectionMetadataGenerator#createMiddleEntityXml()`, now that we have a proper method.